### DISCUSSION TITLE
We are trying to load the Sample Poland fiscal integration into a Tier 1 VM. We are looking into understanding how the provided sample fiscal solution works?

### DISCUSSION THREAD

**Discussion by Bertila-Antony** [microsoft/Dynamics365Commerce.Solutions#344] [Q&A]
We are trying to load the Sample Poland fiscal integration solution from the GitHub path "https://github.com/microsoft/Dynamics365Commerce.Solutions/tree/release/9.51/src/FiscalIntegration/PosnetThermalFVSample" into a Tier 1 VM. We are looking into understanding how the provided sample fiscal solution works?

We noticed the above sample solution contains a Pos extension trigger file “ExtensionPackageTrigger” in the commerce runtime project. However, there is no POS extension project targeting the same in the mentioned GitHub path. Once we installed the scale unit and the store commerce installers, we can see the extension does not load in the store commerce (Settings screen). Please confirm on why we have the ExtensionPackageTrigger class in the CRT project.
 
Additional Information:
•	Commerce SDK Version: 9.51
•	MSBuild Logs: 
•	Reproduction Steps: Download the above sample solution and install the scale unit and the store commerce installers.
•	Environment Details: Tier 1 VM
•	Expected Behavior: The extensions should load without any error in store commerce.
•	Actual Behavior: The extension does not load in store commerce (settings screen)

**Comment by FedyukovDA** [comment]
@Bertila-Antony 
Thanks for the question. 
That is by design. ExtensionPackageTrigger is required not only to load the POS extensions, but also needed to load CRT extensions in offline mode. 
You may ignore the exception on the Settings screen regarding this extension loading. 

Happy holiday, 
DF

**Reply by Bertila-Antony** [reply]
Thanks for Reply !!! After loading the Sample Poland fiscal integration solution from the GitHub path "https://github.com/microsoft/Dynamics365Commerce.Solutions/tree/release/9.51/src/FiscalIntegration/PosnetThermalFVSample" into a Tier 1 VM, now we are facing an issue of Fiscal register error 
![Screenshot 2024-12-16 161118](https://github.com/user-attachments/assets/8f8810ed-03b1-4999-8110-e45f5fcb609d)
![Screenshot 2024-12-16 161301](https://github.com/user-attachments/assets/ef315d1f-afb3-4768-95b3-7202ce6751ee)

**Comment by FedyukovDA** [comment]
@Bertila-Antony 
If I'm not wrong, this file from the exception "libposcmbth.dll" should be provided by Posnet and can be a part of printer drivers. 
We've got mentions of those files in the [old documentation](https://learn.microsoft.com/en-us/dynamics365/commerce/localizations/dev-itpro/emea-pol-fpi-sample-sdk#hardware-station-extension-components)
But I'm surprised that we don't have it in a new one. I'll open a documentation bug to fix that. 

Meanwhile, I found that dll on Posnet website https://www.posnet.com.pl/biblioteki-dll
But it's better to check what was supplied with your printer or printer emulator and copy those files to the HWS bin folder.

**Reply by Bertila-Antony** [reply]
Is there any sample fiscal registration where the document provider and fiscal connector classes are all available in the same Commerce runtime project?

**Reply by Bertila-Antony** [reply]
When we are logging into POS and conclude the transaction, we are facing the below error, could you please give some guidance on this?
<img width="913" alt="image" src="https://github.com/user-attachments/assets/ad6b5a0b-8ade-4429-bb48-3d31a6a62eaf" />

**Reply by FedyukovDA** [reply]
@Bertila-Antony 

> Is there any sample fiscal registration where the document provider and fiscal connector classes are all available in the same Commerce runtime project?

No, as I know, there are not such examples. And I don't think that this case can be applicable to Poland integration, since the fiscal device is used and communication should be processed through Hardware station. 


> When we are logging into POS and conclude the transaction, we are facing the below error, could you please give some guidance on this?

The error basically says that fiscal device is not visible to the pos. 
Verify with the [guide ](https://learn.microsoft.com/en-us/dynamics365/commerce/localizations/dev-itpro/emea-pol-fpi-sample#set-up-the-registration-process)that all steps are completed. Also, Hardware profile to which fiscal device is connected should be selected for the current shift.

**Comment by Bertila-Antony** [comment]
We were able to load the provided Poland specific fiscal integration sample into our local Tier 1 VM by utilizing the CRT project provided in the https://urldefense.com/v3/__https:/github.com/microsoft/Dynamics365Commerce.Solutions/tree/release/9.51/src/FiscalIntegration/PosnetThermalFVSample__;!!OrxsNty6D4my!6UVQM8ff1LH0DhRayEkIcRx5R3t184RBZUJRbcytEy9Pcr_SrQter7ChDo_yAuh8_HIgdWirpRCWd5vrQtkKBgE8s1wSdBed1Q$
When we concluded a transaction in POS, we noticed that the sales transaction is created into the Channel DB before the fiscal registration or process is initiated.
However, we require the system to not create the sales transaction into DB unless the fiscal registration or process succeeds.

Please confirm on whether it is possible to restrict the creation of the sales transaction if the fiscal registration or process fails.
